Sources/SWBTaskExecution/TaskActions/PrecompileClangModuleTaskAction.swift

```diff
@@ -215,16 +215,30 @@ final public class PrecompileClangModuleTaskAction: TaskAction, BuildValueValida
                         enableStrictCASErrors: key.casOptions!.enableStrictCASErrors
                     )
                 }
-            } else if result == .failed && !executionDelegate.userPreferences.enableDebugActivityLogs && !executionDelegate.emitFrontendCommandLines {
-                let commandString = UNIXShellCommandCodec(
-                    encodingStrategy: .backslashes,
-                    encodingBehavior: .fullCommandLine
-                ).encode(commandLine)
-
-                // <rdar://59354519> We need to find a way to use the generic infrastructure for displaying the command line in
-                // the build log.
-                outputDelegate.emitOutput("Failed frontend command:\n")
-                outputDelegate.emitOutput(ByteString(encodingAsUTF8: commandString) + "\n")
+            } else if result == .failed {
+                if !executionDelegate.userPreferences.enableDebugActivityLogs && !executionDelegate.emitFrontendCommandLines {
+                    let commandString = UNIXShellCommandCodec(
+                        encodingStrategy: .backslashes,
+                        encodingBehavior: .fullCommandLine
+                    ).encode(commandLine)
+
+                    // <rdar://59354519> We need to find a way to use the generic infrastructure for displaying the command line in
+                    // the build log.
+                    outputDelegate.emitOutput("Failed frontend command:\n")
+                    outputDelegate.emitOutput(ByteString(encodingAsUTF8: commandString) + "\n")
+                }
+                if case .some(.exit(.uncaughtSignal, _)) = outputDelegate.result {
+                    do {
+                        if let reproducerMessage = try clangModuleDependencyGraph.generateReproducer(
```

Comment:
Do you have a PR for how `generateReproducer` is implemented? I don't like a in-process generator reproducer because it is likely to crash again and take down the entire build system, e.g. it is a crash in preprocessor.

Reply:
I see. This reproducer has no preprocessing steps more than the dependency scanning which is probably fine even I might still prefer an out of process tool for that, but that will need a bigger change to squeeze multiple tool execution into one task.